### PR TITLE
[WIP] feat(xgboost): load xgboost lazily as required

### PR DESCRIPTION
**Pull request by sauyon**
Also modify `LazyType` docs a bit.

### CODE REVIEW

File: bentoml/_internal/frameworks/xgboost.py [hunk 1]
```diff
@@ -91,6 +100,8 @@ def load_model(bento_model: str | Tag | bentoml.Model) -> xgb.core.Booster:
             f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
         )
 
+    xgb = import_xgboost()
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/xgboost.py:103]:
```suggestion
    if not TYPE_CHECKING:
        xgb = import_xgboost()
```

**Comment by ssheng** [comment]
@sauyon do you think this PR is needed for 1.0?

**Comment by sauyon** [comment]
Nope.

File: bentoml/_internal/frameworks/xgboost.py [hunk 2]
```diff
@@ -197,6 +208,8 @@ class XGBoostRunnable(bentoml.Runnable):
 
         def __init__(self):
             super().__init__()
+            self.xgb = import_xgboost()
```
Comment (aarnphm) [inline bentoml/_internal/frameworks/xgboost.py:211]:
```suggestion
            if not TYPE_CHECKING:
                self.xgb = import_xgboost()
```

**Pull request closed without merging**